chain/network/src/peer/codec.rs

```diff
@@ -78,9 +78,7 @@ impl Decoder for Codec {
             return Ok(None);
         }
 
-        let mut len_bytes: [u8; 4] = [0; 4];
-        len_bytes.copy_from_slice(&buf[0..4]);
-        let len = u32::from_le_bytes(len_bytes);
+        let len = buf.get_u32_le();
```

Comment:
Wait, I think this is not actually buggy -- `get_u32_le` actually advances the buffer. That is, we should change the code bellow, `buf[4..4 + len as usize].to_vec()` to be just  `but.split_of(len).to_vec()`. Do we have any failing tests in this PR? If we don't we miss some!

Reply:
@matklad Yes, a lot of tests are failing.

Actually, consuming here is no a good idea. We first peek at the value, and then we try to determine whenever we have read enough bytes to read the whole message. We don't want to consume if that didn't happen.

We could make the Codec into a state machine, which remembers last length read, but I would rather avoid it.

I replaced the code above with
```rust
        let len = u32::from_le_bytes([buf[0], buf[1], buf[2], buf[3]]);
```
for now.

Reply:
@matklad Btw, this `to_vec()` is inefficient. I already have plans for it: https://github.com/near/nearcore/pull/5605